deltachat-contact-tools/src/lib.rs

```diff
@@ -29,10 +29,116 @@ use std::fmt;
 use std::ops::Deref;
 
 use anyhow::bail;
+use anyhow::format_err;
+use anyhow::Context as _;
 use anyhow::Result;
+use chrono::DateTime;
 use once_cell::sync::Lazy;
 use regex::Regex;
 
+// TODOs to clean up:
+// - Check if sanitizing is done correctly everywhere
+// - Apply lints everywhere (https://doc.rust-lang.org/cargo/reference/workspaces.html#the-lints-table)
+
+#[derive(Debug)]
+/// A Contact, as represented in a VCard.
+pub struct VcardContact {
+    /// The email address, vcard property `email`
+    pub addr: String,
+    /// The contact's display name, vcard property `fn`
+    pub display_name: String,
+    /// The contact's public PGP key, vcard property `key`
+    pub key: Option<String>,
+    /// The contact's profile photo (=avatar), vcard property `photo`
+    pub profile_photo: Option<String>,
+    /// The timestamp when the vcard was created / last updated, vcard property `rev`
+    pub timestamp: Result<u64>,
+}
+
+pub fn parse_vcard(vcard: String) -> Result<Vec<VcardContact>> {
+    fn remove_prefix<'a>(s: &'a str, prefix: &str) -> Option<&'a str> {
+        let start_of_s = s.get(..prefix.len())?;
+
+        if start_of_s.eq_ignore_ascii_case(prefix) {
+            s.get(prefix.len()..)
+        } else {
+            None
+        }
+    }
+    fn vcard_property<'a>(s: &'a str, property: &str) -> Option<&'a str> {
+        let remainder = remove_prefix(s, property)?;
+
+        // TODO this doesn't handle the case where there are quotes around a colon
+        let (_params, value) = remainder.split_once(':')?;
```

Comment:
> Shouldn't `_params` be empty?

No, e.g. in the line `EMAIL;TYPE=work:alice@example.com`, `_params` is `;TYPE=work`. But we could probably check whether it's either empty or starts with a `;` (not sure if there is some other special character with which it could start, too)

Reply:
Maybe just check that `_params` doesn't start with any of `[a-zA-Z0-9_]`?

Reply:
Used `char::is_ascii_punctuation()` for this, but excluded "_" as it's usually has the same meaning as alphanumeric chars. Maybe it's better to dive into vCard standards/RFCs, but then we need also check relevant implementations for conformance